src/browser/api_protocol/api_handlers/service_middleware.ts

```diff
@@ -64,7 +64,10 @@ function setTargetIdentity(identity: Identity, payload: any): TargetIdentity {
         return { targetIdentity, serviceIdentity };
     }
     // Sender could be service or client, want service Identity sent in payload either way
-    const serviceIdentity = Service.getServiceByUuid(uuid) || Service.getServiceByUuid(identity.uuid);
+    let serviceIdentity = Service.getServiceByUuid(uuid) || Service.getServiceByUuid(identity.uuid);
```

Comment:
Do we need to be concerned if the two entities that are talking are both registered services? Can this get into a state where it resolves the incorrect one?